compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/TileAttention.cpp

```diff
@@ -375,32 +375,36 @@ void convertToOnlineAttention(IREE::LinalgExt::AttentionOp attnOp,
   }
   AffineMap sumMap = maxMap;
 
-  SmallVector<Range> sizes = attnOp.getIterationDomain(rewriter);
+  AffineMap accMap = attnOp.getOutputMap();
+
+  SmallVector<Range> domain = attnOp.getIterationDomain(rewriter);
 
   // Create fill for acc, max and sum.
   // TODO: Acc should not need a fill. The attention op should get a filled
   // input instead of an empty input.
-  Value zeroAcc = rewriter.create<arith::ConstantOp>(
-      loc, rewriter.getZeroAttr(attnOp.getOutputType().getElementType()));
-  Value accFill =
-      rewriter
-          .create<linalg::FillOp>(loc, ValueRange{zeroAcc}, attnOp.getOutput())
-          .result();
 
+  SmallVector<OpFoldResult> sizes =
+      llvm::map_to_vector(domain, [](Range x) { return x.size; });
+  SmallVector<OpFoldResult> accSize =
+      applyPermutationMap<OpFoldResult>(accMap, sizes);
   SmallVector<OpFoldResult> rowRedSize =
-      llvm::map_to_vector(sizes, [](Range x) { return x.size; });
-  rowRedSize = applyPermutationMap<OpFoldResult>(maxMap, rowRedSize);
+      applyPermutationMap<OpFoldResult>(maxMap, sizes);
 
   Type f32Type = rewriter.getF32Type();
+  Value acc = rewriter.create<tensor::EmptyOp>(loc, accSize, f32Type);
   Value rowRedEmpty =
       rewriter.create<tensor::EmptyOp>(loc, rowRedSize, f32Type);
 
+  Value accInit = rewriter.create<arith::ConstantOp>(
+      loc, rewriter.getZeroAttr(attnOp.getOutputType().getElementType()));
```

Comment:
Shouldn't we use `acc.getType().getElementType()` instead of the `attnOp.getOutputType()`?

Reply:
fixed, thanks